use-shopping-cart/core/warning-middleware.js

```diff
@@ -0,0 +1,20 @@
+function checkCountType(count, action) {
+  if (typeof count !== 'number') {
+    console.warn(`${action.type} requires that count be a number`, action)
+  }
+}
+
+export const handleWarnings = (store) => (next) => async (action) => {
+  const count = action.payload?.options?.count
+  const quantity = action.payload?.quantity
+  if (count) {
+    checkCountType(count, action)
+  } else if (quantity) {
+    checkCountType(quantity, action)
+  }
```

Comment:
These conditional statements are missing some edge cases. If `count` or `quantity` are `0`, `false`, `null`, `undefined`, or some other falsey value the condition will fail and the value will be allowed through. I don't think that there is a way to get around this by checking `count` or `quantity` to determine their necessity.

I would suggest checking the action's name `cart/addItem` and others and based on that check their values.

```js
switch (action.type) {
  // These have `count`
  case 'cart/addItem':
  case 'cart/incrementItem':
  case 'cart/decrementItem':
    if (!isValidCount(count, action)) return
    break
  // This one has `quantity`
  case 'setItemQuantity':
    /* do the same for quantity */
    break
  default: // if it's not those, do nothing
    break
}
```

Reply:
The reason for returning from the middleware is so that the invalid `count` or `quantity` isn't passed through.